Libretro core no longer builds due to zstd

Libretro core (commit c4eafcf008caddd168cbcb45f2206f56818e09f7) no longer builds due to zstd.  The issue occurs as a result of https://github.com/hrydgard/ppsspp/pull/14369.

Update: Just noticed the comments in the above PR, apparently this has been identified and is being looked into by @unknownbrackets (will leave the issue report open though for clean tracking).

Test system is linux mint 20.1 and kernel 5.11 with cmake 3.16.3 and gcc 10.2.0

Build log excerpt: 
```
<pre>Serializer.cpp:(.text._ZN16CChunkFileReader8LoadFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPS5_RPhRmS8_+0x2c4): undefined reference to `ZSTD_decompress&apos;
/usr/bin/ld: Serializer.cpp:(.text._ZN16CChunkFileReader8LoadFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPS5_RPhRmS8_+0x2cc): undefined reference to `ZSTD_isError&apos;
/usr/bin/ld: ../Common/Serialize/Serializer.o: in function `CChunkFileReader::SaveFile(std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, char const*, unsigned char*, unsigned long)&apos;:
Serializer.cpp:(.text._ZN16CChunkFileReader8SaveFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES7_PKcPhm+0x9f): undefined reference to `ZSTD_compressBound&apos;
/usr/bin/ld: Serializer.cpp:(.text._ZN16CChunkFileReader8SaveFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES7_PKcPhm+0xd9): undefined reference to `ZSTD_compress&apos;
/usr/bin/ld: ../GPU/Debugger/Playback.o: in function `GPURecord::ReadCompressed(unsigned int, void*, unsigned long, unsigned int)&apos;:
Playback.cpp:(.text._ZN9GPURecordL14ReadCompressedEjPvmj+0xca): undefined reference to `ZSTD_decompress&apos;
/usr/bin/ld: ../GPU/Debugger/Record.o: in function `GPURecord::WriteCompressed(_IO_FILE*, void const*, unsigned long)&apos;:
Record.cpp:(.text._ZN9GPURecordL15WriteCompressedEP8_IO_FILEPKvm+0x2a): undefined reference to `ZSTD_compressBound&apos;
/usr/bin/ld: Record.cpp:(.text._ZN9GPURecordL15WriteCompressedEP8_IO_FILEPKvm+0x4f): undefined reference to `ZSTD_compress&apos;
/usr/bin/ld: ../Common/Data/Format/ZIMLoad.o: in function `LoadZIMPtr(unsigned char const*, unsigned long, int*, int*, int*, unsigned char**)&apos;:
ZIMLoad.cpp:(.text._Z10LoadZIMPtrPKhmPiS1_S1_PPh+0x21c): undefined reference to `ZSTD_decompress&apos;
collect2: error: ld returned 1 exit status
make: *** [Makefile:624: ppsspp_libretro.so] Error 1
</pre>
```

--- a/libretro/Makefile.common
+++ b/libretro/Makefile.common
@@ -6,6 +6,7 @@ GPUCOMMONDIR		= $(CORE_DIR)/GPU/Common
 GPUDIR			= $(CORE_DIR)/GPU
 EXTDIR			= $(CORE_DIR)/ext
 NATIVEDIR		= $(EXTDIR)/native
+ZSTDDIR                 = $(EXTDIR)/zstd/lib
 
 SOURCES_CXX   :=
 SOURCES_C     :=
@@ -18,6 +19,7 @@ INCFLAGS += \
 	-I$(EXTDIR)/glslang \
 	-I$(EXTDIR)/native \
 	-I$(EXTDIR)/zlib \
+	-I$(ZSTDDIR) \
 	-I$(EXTDIR)/snappy \
 	-I$(FFMPEGDIR) \
 	$(FFMPEGINCFLAGS) \
@@ -700,6 +702,40 @@ SOURCES_CXX += \
 		$(LIBRETRODIR)/LibretroGraphicsContext.cpp \
 		$(LIBRETRODIR)/LibretroGLContext.cpp \
 		$(LIBRETRODIR)/LibretroGLCoreContext.cpp
+
+SOURCES_C +=   $(ZSTDDIR)/common/debug.c \
+               $(ZSTDDIR)/common/entropy_common.c \
+               $(ZSTDDIR)/common/error_private.c \
+               $(ZSTDDIR)/common/fse_decompress.c \
+               $(ZSTDDIR)/common/pool.c \
+               $(ZSTDDIR)/common/threading.c
+
+#SOURCES_C +=   $(ZSTDDIR)/common/xxhash.c
+
+SOURCES_C +=   \
+               $(ZSTDDIR)/common/zstd_common.c \
+               $(ZSTDDIR)/common/zstd_trace.c \
+               $(ZSTDDIR)/compress/fse_compress.c \
+               $(ZSTDDIR)/compress/hist.c \
+               $(ZSTDDIR)/compress/huf_compress.c \
+               $(ZSTDDIR)/compress/zstd_compress.c \
+               $(ZSTDDIR)/compress/zstd_compress_literals.c \
+               $(ZSTDDIR)/compress/zstd_compress_sequences.c \
+               $(ZSTDDIR)/compress/zstd_compress_superblock.c \
+               $(ZSTDDIR)/compress/zstd_double_fast.c \
+               $(ZSTDDIR)/compress/zstd_fast.c \
+               $(ZSTDDIR)/compress/zstd_lazy.c \
+               $(ZSTDDIR)/compress/zstd_ldm.c \
+               $(ZSTDDIR)/compress/zstd_opt.c \
+               $(ZSTDDIR)/compress/zstdmt_compress.c \
+               $(ZSTDDIR)/decompress/huf_decompress.c \
+               $(ZSTDDIR)/decompress/zstd_ddict.c \
+               $(ZSTDDIR)/decompress/zstd_decompress.c \
+               $(ZSTDDIR)/decompress/zstd_decompress_block.c \
+               $(ZSTDDIR)/dictBuilder/cover.c \
+               $(ZSTDDIR)/dictBuilder/divsufsort.c \
+               $(ZSTDDIR)/dictBuilder/fastcover.c \
+               $(ZSTDDIR)/dictBuilder/zdict.c
 
 ifneq ($(STATIC_LINKING), 1)
 SOURCES_C += \

